src/Cardano/Wallet/DB.hs

```diff
@@ -62,13 +94,18 @@ data DBLayer m s = DBLayer
         :: PrimaryKey WalletId
         -> m (Map (Hash "Tx") (Tx, TxMeta))
         -- ^ Fetch the current transaction history of a known wallet. Returns an
-        -- empty map if the wallet isn't found.
+        -- empty map if the wallet istn't found.
     }
 
--- | Error while trying to insert transaction history in the DB.
-newtype ErrPutTxHistory
-    = ErrNoSuchWallet WalletId
-    deriving (Show, Eq)
+-- | Can't perform given operation because there's no wallet
+newtype ErrNoSuchWallet (operation :: Symbol)
```

Comment:
neat way to encode additional info to which operation the error is related :100: